elastic/apm-server
# Replace package utility with package netutil

## Motivation/summary

Package netutil is focused on network-related utility functions, currently all related to splitting and parsing addresses, and extracting addresses from common HTTP proxy headers.

Server logs now set `source.address` to the immediate network peer's IP address, without the port. If the request has X-Forwarded-For, etc., then the log message may also include `client.ip` as well. Previously we set `source.address` to either the immediate network peer's IP:port, or the originating client IP based on XFF etc. The change brings server logs into line with what we set in trace events.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server
2. Use curl to send some requests with and without an X-Forwarded-For header
3. Check the `source.address` field matches the IP address of the host
4. Check the `client.ip` field is only included in requests with X-Forwarded-For, and its value matches the header

## Related issues

None
